### DISCUSSION TITLE
Can't see a created new view ?

### DISCUSSION THREAD

**Discussion by journeytosilius** [perspective-dev/perspective#1876] [General]
This is what I'm doing to create a new view as per the documentation, but once I load the viewer I can't do any group by, it's grayed out ... what I'm doing wrong ?

```
this.table = perspective.worker().table(this.buffer)

this.table.then(value => {
    value.view({
        group_by: ["buys"],
        columns: ["buys"],
        aggregates: {"buys": "dominant"},
        sort: [["buys", "asc"]]
    });

}).catch(err => {
    console.log(err);
});
```
Thanks for moving this to the discussions area ... note that the table is being updated correctly, the problem is that I can't somehow access the new view

**Comment by texodus** [comment]
@chromafunk Please take another look at the [docs](https://perspective.finos.org/docs/md/js.html), I'm pretty confused at what you expect to happen here.

A `View()` is a query object against a table, if you want to access it you need to do something with it like call a method on it to return a data window e.g. `.to_json()`.  You also mention wanting to see this in a "viewer" - but the code above does not even reference a viewer instance?  The code posted above creates a `View()`, then just lets it go out of scope (which is a [memory leak](https://perspective.finos.org/docs/md/js.html#deleting-a-table-or-view) in Perspective, incidentally).
